projects/kit/components/input-phone-international/input-phone-international.component.ts

```diff
@@ -125,6 +128,12 @@ export class TuiInputPhoneInternationalComponent
         // recalculates mask inside inputPhone to prevent isoCode conflict
         this.changeDetectorRef.detectChanges();
 
+        const maxLength = this.getMaxAllowedLength(isoCode);
+
+        if (this.value.length > maxLength) {
```

Comment:
This piece is not clear to me. Could you explain it please?

Reply:
I think we should cut value if the new country mask is shorter than the current value. If it was ten-digit phone number in control and we select country with nine-digit mask, for example. maxAllowedLength is length of the new mask